documentation/course/e2/README.md

```diff
@@ -732,45 +749,96 @@ TODO @jakubzehner: porównanie z NoSQL
+        <li>Mniejsza kontrola</li>
+        <li>Limitacje usługi</li>
       </ul>
     </td>
   </tr>
 </table>
 
-**Decyzja:**
+**Decyzja:** Ze względu na wymagania dotyczące integralności, niezawodności i bezpieczeństwa danych, zdecydowano się na zastosowanie **relacyjnych baz danych ACID** na **Amazon Relational Database Service (RDS)**.
 
-**Opis:**
+**Opis:** Zastosowanie baz danych zgodnych z modelem ACID (Atomicity, Consistency, Isolation, Durability) na platformie AWS RDS (Relational Database Service) jest rozwiązaniem przeznaczonym dla systemów wymagających spójności i niezawodności danych w krytycznych operacjach. Bazy ACID, takie jak MySQL, PostgreSQL, Oracle czy SQL Server, zapewniają gwarancję, że każda transakcja zostanie przetworzona w sposób kompletny i spójny, nawet w przypadku awarii.
 
-<!--
-Głównymi modelami transakcji w bazach danych są podejścia ACID i BASE. Model ACID znany głównie z baz relacyjnych, skupia się na zapewnieniu spójności, poprzez właściwości atomowości, spójności, izolacji i trwałości. W podejściu BASE, stosowanym w bazach NoSQL, poświęcamy spójność na rzecz dostępności, gdzie spójność danych jest osiągana w pewnym czasie, a nie natychmiast[^acid-base].
+Wybierając RDS jako platformę dla baz ACID, zyskuje się zautomatyzowane zarządzanie infrastrukturą, obejmujące tworzenie kopii zapasowych, aktualizacje systemu i konfigurację replikacji. AWS RDS obsługuje także funkcję Multi-AZ (Multi-Availability Zone), która zapewnia wysoką dostępność i odporność na awarie. W przypadku awarii instancji głównej, RDS automatycznie przełącza ruch na zapasowy węzeł w innej strefie dostępności, minimalizując przestoje. Dodatkowo, usługa pozwala na łatwe skalowanie zarówno w pionie (zwiększenie zasobów instancji), jak i poziomie (dodanie replik odczytu), co sprawia, że baza danych może być dostosowana do zmieniających się potrzeb aplikacji.
 
-W przypadku danych kont, bardziej pożądane właściwości ma **model ACID** - istotne jest natychmiastowe odzwierciedlenie zmian w bazie danych, np. dla zmiany hasła użytkownika. Istotna jest również spójność danych z regułami biznesowymi w każdym momencie, np. w przypadku unikalności adresu e-mail. W związku z tym, zdecydowano się na zastosowanie bazy relacyjnej **SQL**. Wadą takiego rozwiązania jest niższa skalowalność horyzontalna w porównaniu do baz NoSQL, jednakże nie powinno to stanowić problemu w serwisie odpowiedzialnym za konta użytkowników.
--->
+AWS RDS eliminuje konieczność ręcznego zarządzania serwerem, pozwalając skupić się na projektowaniu bazy danych i optymalizacji zapytań. Dzięki natywnemu wsparciu dla ACID, zapewnia gwarancje transakcyjne i spójność danych, co czyni je odpowiednim wyborem dla aplikacji wymagających niezawodności oraz bezpieczeństwa. Rozwiązanie to łączy stabilność i wydajność baz transakcyjnych z wygodą korzystania z usług zarządzanych w chmurze, jednocześnie redukując ryzyko błędów administracyjnych oraz koszty związane z utrzymaniem infrastruktury.
 
```

Comment:
**suggestion:** I'd describe why we decided to use SQL everywhere (as opposed to deciding on a service to service basis). I still personally believe that Przemek's service would be better off on NoSQL, so let's give some arguments for not doing this (e.g. "for consistency").